spec/features/recipes/show_spec.rb

```diff
@@ -0,0 +1,89 @@
+
+  let!(:recipe_ingredient_1) { RecipeIngredient.create!(recipe: recipe_1, ingredient: ingredient_1) }
+  let!(:recipe_ingredient_2) { RecipeIngredient.create!(recipe: recipe_1, ingredient: ingredient_2) }
+  let!(:recipe_ingredient_3) { RecipeIngredient.create!(recipe: recipe_1, ingredient: ingredient_3) }
+  let!(:recipe_ingredient_4) { RecipeIngredient.create!(recipe: recipe_1, ingredient: ingredient_4) }
+
+  let!(:recipe_ingredient_5) { RecipeIngredient.create!(recipe: recipe_2, ingredient: ingredient_2) }
+  let!(:recipe_ingredient_6) { RecipeIngredient.create!(recipe: recipe_2, ingredient: ingredient_3) }
+
+  describe "recipe 1 show page" do
+    before(:each) do
+      visit "/recipes/#{recipe_1.id}"
+    end
+
+    it "lists the recipe's name, complexity, and genre" do
+      expect(page).to have_content(recipe_1.name)
+      expect(page).to have_content("Complexity: #{recipe_1.complexity}")
+      expect(page).to have_content("Genre: #{recipe_1.genre}")
+      expect(page).to_not have_content(recipe_2.name)
+    end
+
+    it "lists the recipe's ingredients" do
+      expect(page).to have_content("Ingredients:")
+
+      within "#ingredients" do
+        expect(page).to have_content(ingredient_1.name)
+        expect(page).to have_content(ingredient_2.name)
+        expect(page).to have_content(ingredient_3.name)
+        expect(page).to have_content(ingredient_4.name)
+      end
+    end
+
+    it "lists the total cost of the recipe" do
+      expect(page).to have_content("Total Cost: $7")
+    end
+  end
+
+  describe "recipe 2 show page" do
+    before(:each) do
+      visit "/recipes/#{recipe_2.id}"
+    end
+
+    it "lists the recipe's name, complexity, and genre" do
+      expect(page).to have_content(recipe_2.name)
+      expect(page).to have_content("Complexity: #{recipe_2.complexity}")
+      expect(page).to have_content("Genre: #{recipe_2.genre}")
+      expect(page).to_not have_content(recipe_1.name)
+    end
+
+    it "lists the recipe's ingredients" do
+      expect(page).to have_content("Ingredients:")
+
+      within "#ingredients" do
+        expect(page).to_not have_content(ingredient_1.name)
+        expect(page).to have_content(ingredient_2.name)
+        expect(page).to have_content(ingredient_3.name)
+        expect(page).to_not have_content(ingredient_4.name)
+      end
+    end
+
+    it "lists the total cost of the recipe" do
+      expect(page).to have_content("Total Cost: $3")
+    end
+
+    it "displays a form to add an ingredient to the recipe" do
+      expect(page).to have_content("Add an Ingredient to this Recipe by Ingredient ID")
+      expect(page).to have_field(:ingredient_id)
+      expect(page).to have_button("Submit")
+
+      fill_in :ingredient_id, with: ingredient_1.id
+      click_button "Submit"
+
+      expect(current_path).to eq("/recipes/#{recipe_2.id}")
+
+      expect(page).to have_content(ingredient_1.name)
+      expect(page).to have_content("Total Cost: $6")
```

Comment:
We would expect the calculation to update when we add an ingredient (assuming we coded it correctly), but I like that you're verifying we see that change here as well. There are ways that code could be altered such that it wouldn't be a guarantee, and this test would catch that if such a change were ever made.